### PR TITLE
Don't fetch SCTs when mirroring.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: cpp/fetcher/peer_group.h [hunk 1]
```diff
@@ -22,7 +22,7 @@ namespace cert_trans {
 // TODO(pphaneuf): Make that last sentence true!
 class PeerGroup {
  public:
-  PeerGroup() = default;
+  PeerGroup(bool fetch_scts_);
```
Comment (pphaneuf) [inline cpp/fetcher/peer_group.h]: Explicit?

File: cpp/log/cluster_state_controller.h [hunk 2]
```diff
@@ -103,6 +104,7 @@ class ClusterStateController {
   void ClusterServingSTHUpdater();
 
   const std::shared_ptr<libevent::Base> base_;
+  const bool is_mirror_;                  // This log instance is a mirror
```
Comment (pphaneuf) [inline cpp/log/cluster_state_controller.h]: This is never read? :smiley:

File: cpp/fetcher/peer_group.h (hunk 1 above)
Comment (AlCutter) [inline cpp/fetcher/peer_group.h]: Done

File: cpp/server/handler.cc [hunk 3]
```diff
@@ -509,6 +509,8 @@ void HttpHandler::BlockingGetEntries(evhttp_request* req, int64_t start,
         !cert.SerializeExtraData(&extra_data) ||
         (include_scts &&
          Serializer::SerializeSCT(cert.sct(), &sct_data) != Serializer::OK)) {
+      LOG(WARNING) << "Failed to serialize entry @ " << i;
+      LOG(WARNING) << cert.DebugString();
```
Comment (pphaneuf) [inline cpp/server/handler.cc]: Combine in a single log line, with a newline?

File: cpp/fetcher/peer_group.cc [hunk 4]
```diff
@@ -43,6 +43,9 @@ void GetEntriesDone(AsyncLogClient::Status client_status,
 }  // namespace
 
 
+PeerGroup::PeerGroup(bool fetch_scts) : fetch_scts_(fetch_scts) {
+}
+
```
Comment (pphaneuf) [inline cpp/fetcher/peer_group.cc]: One more line here.

**Comment by pphaneuf** [comment]
Done for this round.

File: cpp/log/cluster_state_controller.h (hunk 2 above)
Comment (AlCutter) [inline cpp/log/cluster_state_controller.h]: Ah yes, this was originally used for some fiddling in `DetermineElectionParticipation` which I removed when I realised that whole method could bite the dust.
Dinged.

File: cpp/server/handler.cc (hunk 3 above)
Comment (AlCutter) [inline cpp/server/handler.cc]: Done.

File: cpp/fetcher/peer_group.cc (hunk 4 above)
Comment (AlCutter) [inline cpp/fetcher/peer_group.cc]: Added

**Comment by AlCutter** [comment]
PTAL

**Comment by pphaneuf** [comment]
LGTM

**Pull request merged**